flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/ResolvedSchema.java

```diff
@@ -63,15 +63,25 @@ public final class ResolvedSchema {
     private final List<Column> columns;
     private final List<WatermarkSpec> watermarkSpecs;
     private final @Nullable UniqueConstraint primaryKey;
+    @Deprecated private final List<Index> indexes;
```

Comment:
We should not introduce a new public interface with `Deprecated`. `Deprecated` means we will delete it in future.